crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -55,11 +55,30 @@ impl<'w, E, B: Bundle> Trigger<'w, E, B> {
         Ptr::from(&self.event)
     }
 
-    /// Returns the entity that triggered the observer, could be [`Entity::PLACEHOLDER`].
+    /// Returns the entity that triggered the observer.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the entity is [`Entity::PLACEHOLDER`].
+    /// Use [`Trigger::get_entity`] if you want to handle the case where the entity is a placeholder.
     pub fn entity(&self) -> Entity {
+        if self.trigger.entity == Entity::PLACEHOLDER {
+            panic!(
+                "called `Trigger::entity()` when the target entity was a `Entity::PLACEHOLDER`."
```

Comment:
This message does not help a user, as they will mostly not interact with `Entity::PLACEHOLDER` and probably got this panic while trying to get the entity of a globally triggered observer. Add a hint to this like "Did you try [...]? There is no associated entity in that case."

This comment is also valid for the `# Panics` documentation, which should mention under which circumstances an underlying `Entity` would be a placeholder.